pkg/storage/storagewrappers/boundedconcurrency.go

```diff
@@ -94,7 +95,13 @@ func (b *boundedConcurrencyTupleReader) waitForLimiter(ctx context.Context) {
 
 	end := time.Now()
 	timeWaiting := end.Sub(start).Milliseconds()
-	boundedReadDelayMsHistogram.Observe(float64(timeWaiting))
+
+	rpcInfo := telemetry.RPCInfoFromContext(ctx)
+	boundedReadDelayMsHistogram.WithLabelValues(
+		rpcInfo.Service,
+		rpcInfo.Method,
+	).Observe(float64(timeWaiting))
```

Comment:
don't we need to add `if rpcInfo != nil` here?

Reply:
Yes.  It is added